### PR TITLE
multiple api paths support and updates

### PR DESCRIPTION
**Pull request by therfert**
##### SUMMARY
new features:
 - support API paths ``ipv6 address``, ``ipv6 dhcp-server``, ``ipv6 dhcp-server option``, ``ipv6 route``, ``queue tree``, ``routing ospf area``, ``routing ospf area range``, ``routing ospf instance``, ``routing ospf interface-template``, ``routing pimsm instance``, ``routing pimsm interface-template``
- support for fields ``blackhole``, ``pref-src``, ``routing-table``, ``suppress-hw-offload``, ``type``, ``vrf-interface`` in ``ip route`` path

bugfixes:
  - ``queue interface`` path works with ``api_modify`` now
  - ``ip route`` entry can be defined wihtout the need of ``gateway`` field (which is correct for unreachable/blackhole type of routes).

##### ISSUE TYPE
- Bugfix Pull Request
- Feature Pull Request

##### COMPONENT NAME
- api_info
- api_modify

##### ADDITIONAL INFORMATION
none

### CODE REVIEW

**Comment by therfert** [comment]
Hi @felixfontein, do you prefer such PR or rather multiple smaller ones as I've been doing before?

File: changelogs/fragments/131-api.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    (https://github.com/ansible-collections/community.routeros/pull/131)
+  - api_modify, api_info - support for fields ``blackhole``, ``pref-src``, ``routing-table``, ``suppress-hw-offload``, ``type``, ``vrf-interface`` in ``ip route`` path
+    (https://github.com/ansible-collections/community.routeros/pull/131)
+known_issues:
+  - api_modify - when limits for entries in ``queue tree`` are defined as human readable - e.g. 25M, the configuration will be correctly set in ROS, but the module will indicate the item is changed on every run even when there was no change done. This is caused by the ROS API which returns the number in bytes - e.g. 25000000 (which is inconsistent with the CLI behavior). In order to mitigate that, the limits have to be defined in bytes (those will still appear as human readable in the ROS CLI).
+    (https://github.com/ansible-collections/community.routeros/pull/131
```
Comment (felixfontein) [inline changelogs/fragments/131-api.yml:8]:
```suggestion
    (https://github.com/ansible-collections/community.routeros/pull/131).
```

File: changelogs/fragments/131-api.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+minor_changes:
+  - api_modify, api_info - support API paths ``ipv6 address``, ``ipv6 dhcp-server``, ``ipv6 dhcp-server option``, ``ipv6 route``, ``queue tree``, ``routing ospf area``, ``routing ospf area range``, ``routing ospf instance``, ``routing ospf interface-template``, ``routing pimsm instance``, ``routing pimsm interface-template``
+    (https://github.com/ansible-collections/community.routeros/pull/131)
```
Comment (felixfontein) [inline changelogs/fragments/131-api.yml:3]:
```suggestion
    (https://github.com/ansible-collections/community.routeros/pull/131).
```

File: changelogs/fragments/131-api.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  - api_modify, api_info - support for fields ``blackhole``, ``pref-src``, ``routing-table``, ``suppress-hw-offload``, ``type``, ``vrf-interface`` in ``ip route`` path
+    (https://github.com/ansible-collections/community.routeros/pull/131)
```
Comment (felixfontein) [inline changelogs/fragments/131-api.yml:5]:
```suggestion
    (https://github.com/ansible-collections/community.routeros/pull/131).
```

File: changelogs/fragments/131-api.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+known_issues:
+  - api_modify - when limits for entries in ``queue tree`` are defined as human readable - e.g. 25M, the configuration will be correctly set in ROS, but the module will indicate the item is changed on every run even when there was no change done. This is caused by the ROS API which returns the number in bytes - e.g. 25000000 (which is inconsistent with the CLI behavior). In order to mitigate that, the limits have to be defined in bytes (those will still appear as human readable in the ROS CLI).
+    (https://github.com/ansible-collections/community.routeros/pull/131
+  - "``routing ospf area``, ``routing ospf area range``, ``routing ospf instance``, ``routing ospf interface-template`` paths are not fully implemeted for ROS6 due to the significat changes between ROS6 and ROS7.
```
Comment (felixfontein) [inline changelogs/fragments/131-api.yml:9]:
```suggestion
  - "api_modify, api_info - ``routing ospf area``, ``routing ospf area range``, ``routing ospf instance``, ``routing ospf interface-template`` paths are not fully implemeted for ROS6 due to the significat changes between ROS6 and ROS7
```

File: changelogs/fragments/131-api.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    (https://github.com/ansible-collections/community.routeros/pull/131)"
```
Comment (felixfontein) [inline changelogs/fragments/131-api.yml:10]:
```suggestion
    (https://github.com/ansible-collections/community.routeros/pull/131)."
```

File: changelogs/fragments/131-api.yml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+minor_changes:
+  - api_modify, api_info - support API paths ``ipv6 address``, ``ipv6 dhcp-server``, ``ipv6 dhcp-server option``, ``ipv6 route``, ``queue tree``, ``routing ospf area``, ``routing ospf area range``, ``routing ospf instance``, ``routing ospf interface-template``, ``routing pimsm instance``, ``routing pimsm interface-template``
+    (https://github.com/ansible-collections/community.routeros/pull/131)
+  - api_modify, api_info - support for fields ``blackhole``, ``pref-src``, ``routing-table``, ``suppress-hw-offload``, ``type``, ``vrf-interface`` in ``ip route`` path
+    (https://github.com/ansible-collections/community.routeros/pull/131)
+known_issues:
+  - api_modify - when limits for entries in ``queue tree`` are defined as human readable - e.g. 25M, the configuration will be correctly set in ROS, but the module will indicate the item is changed on every run even when there was no change done. This is caused by the ROS API which returns the number in bytes - e.g. 25000000 (which is inconsistent with the CLI behavior). In order to mitigate that, the limits have to be defined in bytes (those will still appear as human readable in the ROS CLI).
```
Comment (felixfontein) [inline changelogs/fragments/131-api.yml:7]:
```suggestion
  - api_modify - when limits for entries in ``queue tree`` are defined as human readable - for example ``25M`` -, the configuration will be correctly set in ROS, but the module will indicate the item is changed on every run even when there was no change done. This is caused by the ROS API which returns the number in bytes - for example ``25000000`` (which is inconsistent with the CLI behavior). In order to mitigate that, the limits have to be defined in bytes (those will still appear as human readable in the ROS CLI)
```

File: changelogs/fragments/131-api.yml [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    (https://github.com/ansible-collections/community.routeros/pull/131
+  - "``routing ospf area``, ``routing ospf area range``, ``routing ospf instance``, ``routing ospf interface-template`` paths are not fully implemeted for ROS6 due to the significat changes between ROS6 and ROS7.
+    (https://github.com/ansible-collections/community.routeros/pull/131)"
+bugfixes:
+  - "``queue interface`` path works with api_modify now
```
Comment (felixfontein) [inline changelogs/fragments/131-api.yml:12]:
```suggestion
  - "api_modify - ``queue interface`` path works now
```

File: changelogs/fragments/131-api.yml [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    (https://github.com/ansible-collections/community.routeros/pull/131)."
+  - "``ip route`` entry can be defined wihtout the need of ``gateway`` field (which is correct for unreachable/blackhole type of routes).
```
Comment (felixfontein) [inline changelogs/fragments/131-api.yml:14]:
```suggestion
  - "api_modify - ``ip route`` entry can be defined without the need of ``gateway`` field, which is correct for unreachable/blackhole type of routes
```

**Review by felixfontein — COMMENTED** [review]
Mostly editorial changes to the changelog :)

**Comment by felixfontein** [comment]
> Hi @felixfontein, do you prefer such PR or rather multiple smaller ones as I've been doing before?

It depends, changes such as #128 and #130 definitely should be their own PR, but others like this can be combined.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@therfert thanks for your contribution!

**Pull request merged**